### PR TITLE
[Cypress] [WIP] Fix E2E and integration Tests for UI

### PR DESCRIPTION
**Pull request by Chadha93**
Signed-off-by: Gaurav Chadha <gauravchadha1676@gmail.com>
**Description**

This PR fixes #

- [x] GraphQL Utils for handling Queries and Mutations.
- [ ] E2E test for service_mesh_configuration_management_spec 
- [ ] Integration test Configuration_filters_spec 
- [x] E2E for userpreference_spec
- [x] Integration for userperference_spec  

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
@Chadha93 if this first done, let's proceed with merge, then follow up on the other two checkmarks subsequently.

File: ui/utils/cypress-gql/graphql-test-utils.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// Utility to match GraphQL mutation based on the operation name
+export const hasOperationName = (req, operationName) => {
+  const { body } = req
+  return (
+    Object.prototype.hasOwnProperty.call(req, 'operationName') && body.operationName === operationName
```
Comment (Chadha93) [inline ui/utils/cypress-gql/graphql-test-utils.js:5]:
```suggestion
    Object.prototype.hasOwnProperty.call(body, 'operationName') && body.operationName === operationName
```

**Comment by Chadha93** [comment]
> @Chadha93 if this first done, let's proceed with merge, then follow up on the other two checkmarks subsequently.

I just had a quick huddle with @Abhishek-kumar09 regarding this, since the new GraphQL endpoints are updated in catalog Server PR for Catalog and UI Pr is pending we need to wait to get that merge, and some more items need to be updated here after that we can test and change accordingly. Then we can proceed with it. 

Found a login issue with filter_spec cypress file which was not allowing to visit `/configuration/filters`, making a fix for that separately.

**Comment by leecalcote** [comment]
@Chadha93, oh, very good.

**Comment by leecalcote** [comment]
@Chadha93, reminder to knock this one out:


> Found a login issue with filter_spec cypress file which was not allowing to visit /configuration/filters, making a fix for that separately.

**Comment by Chadha93** [comment]
> @Chadha93, reminder to knock this one out:
> 
> > Found a login issue with filter_spec cypress file which was not allowing to visit /configuration/filters, making a fix for that separately.

Already done https://github.com/meshery/meshery/pull/6139

**Comment by Abhishek-kumar09** [comment]
It depended on https://github.com/meshery/meshery/pull/6027 since the PR is merged, it needs a rebase to pull those changes, I am not updating the branch myself incase you have any uncomitted changes that may introduce you to some extra challenge time for pushing to the current git history. 

The ask is if you have updated the branch, is the gql interceptors working as expected now?

**Comment by Chadha93** [comment]
> It depended on #6027 since the PR is merged, it needs a rebase to pull those changes, I am not updating the branch myself incase you have any uncomitted changes that may introduce you to some extra challenge time for pushing to the current git history.
> 
> The ask is if you have updated the branch, is the gql interceptors working as expected now?

Updated it now.

**Comment by MarioArriaga92** [comment]
beforeEach hook in integration/configuration_filters_spec.js is getting a 500 during request to set the provider @Chadha93:
CleanShot 2022-09-06 at 10 04 43@2x

**Comment by MarioArriaga92** [comment]
@Chadha93 for e2e/service_mesh_configuration_management_spec.js the modified request interception alias never gets called by Meshery UI:
```
CypressError: Timed out retrying after 5000ms: `cy.wait()` timed out waiting `5000ms` for the 2nd request to the route: `getConfigItems`. No request ever occurred.
```

**Comment by Chadha93** [comment]
@leecalcote @Abhishek-kumar09 Cypress checks for Filters are disabled for now and other UI tests are passing.

**Comment by leecalcote** [comment]
Good. We’ll be down a test, but back to green in the meantime.

**Review by leecalcote — APPROVED** [review]

**Comment by Chadha93** [comment]
Merging this one to get green checks on tests. Opening a separate issue for filters test with gql request implementation.

**Pull request merged**